src/main/webapp/scripts/appscript.js

```diff
@@ -17,25 +17,53 @@
   * Customer ID dropdown.
   */
 async function populateCustomerList() {
-  // TODO: Implement functionality for hashmap parsing instead of simple array.
+  // Fetches json dictionary from VendorServlet.
   var response = await fetch('/VendorServlet');
 
-  // Response is a json array that is parsed into json.
-  var customerJson = await response.json();
-
-  console.log(customerJson);
+  var vendorJson = await response.json();
 
   // Loops over this array to make the options for the Customer ID dropdown.
   var vendorHtml = '';
-  for (var i = 0; i < customerJson.length; i++) {
-    html += '<option value="' + customerJson[i] + '">'
-    + customerJson[i] + '</option>';
+  for (var key in vendorJson) {
+    console.log(key);
+    vendorHtml += '<option value="' + key + '">'
+    + key + '</option>';
```

Comment:
Constructing strings in JS could be a little bit cleaner if you used something like [template literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals)